internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -43,6 +45,20 @@ type Receiver struct {
 
 var _ component.MetricsReceiver = (*Receiver)(nil)
 
+var logRedirectsVar logRedirects
+
+func init() {
+	cfg := zap.NewProductionConfig()
+	cfg.Level.SetLevel(zapcore.DebugLevel)
```

Comment:
We want to base everything on the app logger and not our own right?

Reply:
Essentially what tells me to which zap logger to redirect a logrus log entry in Fire() is the monitorType logrus entry field. So what if monitorType is unavailable? Which zap logger do I use? That's why I have that catchall logger.

Reply:
The smart agent shouldn't log anything until a receiver is created so I'm not sure this is a case we need to consider atm.  Having a default logger w/ debug level is probably not desired in general.